examples/08_join_aggregation.py

```diff
@@ -1,302 +1,311 @@
-
-    sns.barplot(x=x, y=y, palette=cmap(norm(y)), ax=ax)
-    ax.set_title(title)
-    ax.set_xticks(ax.get_xticks(), labels=ax.get_xticklabels(), rotation=30)
-    ax.set_ylabel(None)
-
-
-# O is Monday, 6 is Sunday
-
-daily_volume = X_date_encoded["timestamp_weekday"].value_counts().sort_index()
-
-make_barplot(
-    x=daily_volume.index,
-    y=daily_volume.values,
-    title="Daily volume of ratings",
-)
-
-###############################################################################
-# We also display the distribution of our target ``y``.
-rating_count = y.value_counts().sort_index()
-
-make_barplot(
-    x=rating_count.index,
-    y=rating_count.values,
-    title="Distribution of ratings given to movies",
-)
+# Let's explore a naive solution first.
+#
+# The first idea that comes to mind to merge these two tables is to aggregate the
+# products attributes into lists, using their basket IDs.
+products_grouped = products.groupby("basket_ID").agg(list)
+products_grouped
+
+# %%
+# Then, we can expand all lists into columns, as if we were "flattening" the dataframe.
+# We end up with a product dataframe ready to be joined on the basket dataframe, using
+# basket IDs as join keys.
+import pandas as pd
 
+products_flatten = []
+for col in products_grouped.columns:
+    cols = [f"{col}{idx}" for idx in range(24)]
+    products_flatten.append(pd.DataFrame(products_grouped[col].to_list(), columns=cols))
+products_flatten = pd.concat(products_flatten, axis=1)
+products_flatten.insert(0, "basket_ID", products_grouped.index)
+TableReport(products_flatten)
 
-###############################################################################
-# AggTarget: aggregate y, then join
-# ---------------------------------
+# %%
+# Look at the "Stats" section of the ``TableReport`` above. Does anything strike you?
 #
-# We have just extracted datetime features from timestamps.
+# Not only did we create 144 columns, but most of these columns are filled with NaN,
+# which is very inefficient for learning!
 #
-# Let's now perform an expansion for the target ``y``, by aggregating it before
-# joining it back on the main table. The biggest risk of doing target expansion
-# with multiple dataframe operations yourself is to end up leaking the target.
+# This is because each basket contains a variable number of products, up to 24, and we
+# created one column for each product attribute, for each position (up to 24) in
+# the dataframe.
 #
-# To solve this, the |AggTarget| transformer allows you to
-# aggregate the target ``y`` before joining it on the main table, without
-# risk of leaking. Note that to perform aggregation then joining on the features
-# ``X``, you need to use |AggJoiner| instead.
+# Moreover, if we wanted to replace text columns with encodings, we would create
+# :math:`d \times 24 \times 2` columns (encoding of dimensionality :math:`d`, for
+# 24 products, for the "item" and "make" columns), which would explode the memory usage.
```

Comment:
```suggestion
# 24 products, for the ``"item"`` and ``"make"`` columns), which would explode the memory usage.
```

I don't know if we should add this here and below. Pretty minor and fine with it either way